File: fhg-api/src/main/java/de/jbamberger/api/data/AtomFeed.java
package de.jbamberger.api.data;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

/**
 * @author Jannik
 * @version 05.08.2016.
 */
@Root(strict = false)
class AtomFeed {

    @Element(name = "title")
    public String title;

    @Element(name = "subtitle")
    public String subTitle;

    @Element(name = "updated")
    public String updated;

    @Element(name = "icon")
    public String icon;

    @ElementList(name = "entry", inline = true, type = Entry.class)
    public List<Entry> entries;

    @Root(name = "entry")
    public static class Entry {
        @Element(name = "author")
        public Author author;
        @Element(name = "title")
        public String title;
        @Element(name = "link")
        public Link link;
        @Element(name = "id")
        public String id;
        @Element(name = "updated")
        public String updated;
        @Element(name = "published")
        public String published;
        @ElementList(name = "category", type = Category.class, required = false, inline = true)
        public List<Category> categories;
        @Element(name = "summary")
        public String summary;

        public static class Author {
            @Element(name = "name")
            public String name;
        }

    }

    public static class Category {
        @Attribute(name = "scheme")
        public String scheme;
        @Attribute(name = "term")
        public String term;
    }

    public static class Link {
        @Attribute(name = "rel")
        public String rel;
        @Attribute(name = "type")
        public String type;
        @Attribute(name = "href")
        public String href;
    }
}

File: fhg-api/src/main/java/de/jbamberger/api/data/AtomFeed.kt
package de.jbamberger.api.data

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com
 */
@Root(strict = false)
internal class AtomFeed {

    @Element(name = "title")
    var title: String? = null

    @Element(name = "subtitle")
    var subtitle: String? = null

    @Element(name = "updated")
    var updated: String? = null

    @Element(name = "icon")
    var icon: String? = null

    @ElementList(name = "entry", inline = true, type = Entry::class)
    var entries: List<Entry>? = null

    @Root(name = "entry")
    class Entry {
        @Element(name = "author")
        var author: Author? = null
        @Element(name = "title")
        var title: String? = null
        @Element(name = "link")
        var link: Link? = null
        @Element(name = "id")
        var id: String? = null
        @Element(name = "updated")
        var updated: String? = null
        @Element(name = "published")
        var published: String? = null
        @ElementList(name = "category", type = Category::class, required = false, inline = true)
        var categories: List<Category>? = null
        @Element(name = "summary")
        var summary: String? = null

        class Author {
            @Element(name = "name")
            var name: String? = null
        }
    }

    class Category {
        @Attribute(name = "scheme")
        var scheme: String? = null
        @Attribute(name = "term")
        var term: String? = null
    }

    class Link {
        @Attribute(name = "rel")
        var rel: String? = null
        @Attribute(name = "type")
        var type: String? = null
        @Attribute(name = "href")
        var href: String? = null
    }
}
